src/Matrix.php

```diff
@@ -957,6 +957,110 @@ public function matmul(Matrix $b) : self
         return self::quick($c);
     }
 
+    /**
+     * Multiply this matrix with another matrix using the Strassen algorithm (matrix-matrix product).
+     *
+     * @param \Tensor\Matrix $b
+     * @throws \InvalidArgumentException
+     * @return self
+     */
+    public function matmulStrassen(Matrix $b) : self
+    {
+        if ($this->n !== $b->m()) {
+            throw new InvalidArgumentException('Matrix A requires'
+                . " $this->n rows but Matrix B has {$b->m}.");
+        }
+
+        if ($this->n === 1) {
+            return self::quick([[$this->a[0][0] * $b->a[0][0]]]);
+        }
+
+        if ($this->n % 2 !== 0) {
+            $a1 = $this->a;
+            $b1 = $b->a;
+
+            for ($i = 0; $i < $this->n + 1; ++$i) {
+                for ($j = 0; $j < $this->n + 1; ++$j) {
+                    $a1[$i][$j] = $a1[$i][$j] ?? 0;
+                    $b1[$i][$j] = $b1[$i][$j] ?? 0;
+                }
+            }
+
+            $aM = self::quick($a1);
+            $bM = self::quick($b1);
+
+            $c = $aM->matmulStrassen($bM);
+
+            $c1 = [];
+            for ($i = 0; $i < $this->n; ++$i) {
+                for ($j = 0; $j < $this->n; ++$j) {
+                    $c1[$i][$j] = $c->a[$i][$j];
```

Comment:
Can we avoid doubly indexing into the `$c1` array here as well as repeat indexing into the $c->a array (for performance reasons)? Something like this for example?

Ex.

```php
            $c1 = [];
            
            for ($i = 0; $i < $this->n; ++$i) {
                $aRow = $c->a[$i];
              
                $c1Row = [];
            
                for ($j = 0; $j < $this->n; ++$j) {
                    $c1Row[] = $aRow[$j];
               }
               
               $c1[] = $c1Row;
           }
```

While we're benchmarking this new matmul implementation, we can also see what effect this change could bring to the wall time.